### ISSUE TITLE
Model resources contribution

### ISSUE THREAD

**Issue by stevhliu** [huggingface/transformers#20055]
Hi friends! 👋

There are a lot of cool existing resources for how to do *x* with *x* model, and we’d like to showcase and aggregate these resources on a model’s documentation. This’ll help users see how they can get started with a model for their own tasks since we know a lot of users check out the model documentation first. Take a look at a completed [resource section](https://huggingface.co/docs/transformers/main/en/model_doc/distilbert#resources) for DistilBERT as an example.

I’ve identified the top 20 models by pageviews, and now I’d like to open it up to the community if anyone is interested in helping!

Anyone can contribute; you just need to comment and claim one of the models on this [list](https://github.com/huggingface/transformers/issues/19848). Contributing is super easy:

1. Once you've claimed a model from the list, collect the existing resources from:
- the Hugging Face [blog](https://huggingface.co/blog)
- relevant materials from the 🤗 Hugging Face [Course](https://huggingface.co/course/chapter1/1)
- the Hugging Face [example scripts](https://github.com/huggingface/transformers/tree/main/examples) and [notebooks](https://github.com/huggingface/transformers/tree/main/notebooks)
- @NielsRogge's Transformers Tutorials [repository](https://github.com/NielsRogge/Transformers-Tutorials)
- @philschmid's [blog](https://www.philschmid.de/)
- [notebooks](https://huggingface.co/docs/transformers/community) from the community ❤️

2. Organize the resources by model tasks or applications (like inference or deployment):
- Use the corresponding icons for each task (you can find the names for each icon [here](https://github.com/huggingface/doc-builder/blob/19ba9da2556294f1777c865793d13e9ea47f8716/kit/src/lib/PipelineIcon.svelte#L42-L71)):
   ```
   <PipelineTag pipeline=”name-of-task”/>
   ```
- For certain categories, you can just do: 🚀 Deploy, ⚡️ Inference, or ⚗️ Optimization, etc.
- For community resources, add the 🌎 emoji at the end to indicate it’s not an official Hugging Face resource.
- Use this DistilBERT [file](https://github.com/huggingface/transformers/pull/19930/files) as a template. You can copy and paste the intro text and just replace DistilBERT with the name of the model you're working on.
    
3. Open a Pull Request with the new resources for your chosen model and ping me for a review (if you’re just getting started with contributing to an open-source project, check out @merveenoyan's awesome [GitHub Contribution Guide](https://www.notion.so/19411c29298644df8e9656af45a7686d)).

4. Congratulations, you just merged a PR into 🤗 Transformers, and your contribution will now help anyone who is looking at the model docs! 🎉

If you have any questions or need any help, don’t hesitate to ping me! 🤗❤️

**Comment by stevhliu** [comment]
Awesome! I'm looking forward to your contribution, and feel free to ping me if you have any questions! 🤗

**Comment by shogohida** [comment]
@stevhliu 
I have a question. Is there a good way to search GitHub and blog posts? I tried to find related repos and blog posts with `OpenAI GPT` but I couldn't find them because search function doesn't seem to work well... Should I search one by one repo or post?

I made a draft pull request although it doesn't have links of GitHub and blog. You can check it to see if my research has been good or not
https://github.com/huggingface/transformers/pull/20084

**Comment by shogohida** [comment]
@stevhliu 
Thanks for your comment! It will take a lot of time to collect resources from scripts and notebooks because I'm not very familiar with OpenAI GPT but I'll do my best. I'll let you know if I have another question

**Comment by ambujpawar** [comment]
Hi, I would like to take CLIP from the list you have mentioned. :)

**Comment by Saad135** [comment]
@stevhliu I would like to work on DeBERTa

**Comment by stevhliu** [comment]
Great, thanks for taking on DeBERTa @Saad135! 🤗

**Comment by JuheonChu** [comment]
Hello, do you mind if  I can tackle on ALBERT model?

**Comment by stevhliu** [comment]
For sure, looking forward to your contribution @JuheonChu! 🤗

**Comment by stanleycai95** [comment]
Hi! Could I try ViT?

**Comment by hazrulakmal** [comment]
Hi, I would like to work on XLM-RoBERTa! @stevhliu

**Comment by stevhliu** [comment]
Hey @stanleycai95, that would be great! Feel free to work on it when you have the time :)

Awesome, XLM-RoBERTa is all yours @hazrulakmal!

**Comment by adit299** [comment]
Hi, I would like to work on GPT-J!

**Comment by stevhliu** [comment]
Yay thanks for taking on GPTJ @adit299! Let me know if you have any questions or need any help 🤗

**Comment by alissadb** [comment]
Hi, could I work on OPT? :) @stevhliu

**Comment by stevhliu** [comment]
OPT is all yours @alissadb! 🤩

**Comment by Laxmaan** [comment]
Let me round out the list @stevhliu . TrOCR

**Comment by stevhliu** [comment]
Awesome, thanks for finishing this off @Laxmaan! 🎉

**Comment by elabongaatuo** [comment]
Hello @stevhliu . I'd love to contribute in documentation. I see all models are assigned, is there any other I can help with?
Thank you 😊

**Comment by stevhliu** [comment]
Hi @elabongaatuo, sorry for the late reply and thanks for your enthusiasm! 

I think we are good with the model resource contributions for now. If you're looking for ways to contribute to the docs, feel free to open an issue for improving the docs (content that is unclear, missing, or inaccurate or fixing typos) and we can review it there. For more info about getting started with contributing, take a look at this [guide](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md)! 🤗

**Comment by elabongaatuo** [comment]
Hello @stevhliu . Thanks for getting back to me. I'll be on the lookout for docs that need improving.

**Comment by stevhliu** [comment]
Hi @JuheonChu and @Laxmaan, I wanted to check and see if you're still interested in making a model contribution. Totally cool if you aren't available anymore, I'll unassign the models you claimed and let others take a shot at it. Thanks!

**Comment by huangperry** [comment]
Hi @stevhliu, I'd like to take a shot at one of the models if one of them becomes unassigned. Please let me know!

**Comment by stevhliu** [comment]
Thanks for the interest; TrOCR, LayoutLMV2, and ALBERT are now available!

**Comment by elabongaatuo** [comment]
Hello @stevhliu. I'd like to take up  ALBERT.

**Comment by huangperry** [comment]
> Thanks for the interest; TrOCR, LayoutLMV2, and ALBERT are now available!

I’d like to take TrOCR!

**Comment by subham73** [comment]
> Thanks for the interest; TrOCR, LayoutLMV2, and ALBERT are now available!

Hello!! @stevhliu I don't have any option I guess 😅. LayoutLMV2 for me then.

**Comment by stevhliu** [comment]
Hi @subham73, LayoutLMv2 is actually [done](https://github.com/huggingface/transformers/issues/19848) haha!

**Comment by Girish16** [comment]
Hi @stevhliu are there any open issues to work on :)

**Comment by ENate** [comment]
Hi. Is ALBERT still available?

**Comment by stevhliu** [comment]
Hi @ENate, ALBERT is currently being worked on in #23685 (though it has been a while). If the original contributor is no longer interested in working on it, I'll let you know! 😄

**Comment by ENate** [comment]
No worries thanks :) I am looking and navigating your work on hugging face.

**Comment by ENate** [comment]
Okay then. Will proceed using the guidelines provided by @stevhliu and the example for DIstilBERT.

**Comment by stevhliu** [comment]
Yeah ALBERT only has the task guides, and it doesn't go quite as in-depth as DistilBERT. For example, DistilBERT includes links to the course, notebooks, and scripts. You can probably just copy over most of the content from DistilBERT that is relevant to ALBERT (in other words, replace `DistilBERTForX` with `ALBERTForX`)!

**Comment by ENate** [comment]
Thanks :) @stevhliu

**Comment by daniela-basurto** [comment]
Hello @stevhliu is Jukebox still available?

**Comment by stevhliu** [comment]
Feel free to open a PR for Jukebox @daniela-basurto! 🤗

**Comment by wonhyeongseo** [comment]
Hello @stevhliu may I please take up **whisper** with a few of the OSSCA mentees?

**Comment by stevhliu** [comment]
Yes absolutely, thanks for your interest @wonhyeongseo!

**Comment by ahtashamilyas** [comment]
Ok.

**Comment by stevhliu** [comment]
> I'm not sure if all of these are open for contributions though.

Thanks for checking @wonhyeongseo! I think it would be nice to eventually have Resources for all the models, so if you see other ones you're interested in contributing to, feel free to open a PR! I would focus on the more high-impact models first (like LLaMA) that get more pageviews/usage. For certain models (like [BORT](https://huggingface.co/docs/transformers/model_doc/bort)) that are in maintenance mode, we can skip those entirely.

**Comment by wonhyeongseo** [comment]
Awesome @stevhliu , thank you so much for your warm reception. 

- **May we please reserve LLaMA as well for the OSSCA team?**
- **In your opinion, when is the ideal time to start gathering resources after a model's release?**
  For LLaMA2, since it's relatively new, there might not be many official resources yet. It will depend on a models impact as you described, but a rule of thumb would be useful.
- **Although I think this is already the case, would it be possible for you to sort these incomplete models and provide the top 20 sorted by impact or page views as of recent advances?**
  I'm sure some will pique the interest of my team and our mentees.

<summary>

Thank you for the heads up for files under maintenance! I've deleted 8 of those from the above https://github.com/huggingface/transformers/issues/20055#issuecomment-1645188309 list by `grep -LZ "## Resources" * | xargs -0 grep -l "<Tip warning={true}>"` :

</summary>
<details>

```js
auto.md
bort.md
mctct.md
open-llama.md
retribert.md
tapex.md
trajectory_transformer.md
transfo-xl.md
```

</details>

Thank you so much for your support @stevhliu .
Hope you have a wonderful weekend!

Best regards,
Won Seo

**Comment by stevhliu** [comment]
> May we please reserve LLaMA as well for the OSSCA team?

For sure! 👍

> In your opinion, when is the ideal time to start gathering resources after a model's release?

I think maybe whenever you see some content, you can open a PR to add it to the model page. It's ok if it's just one guide/tutorial/blog post; we can gradually add to it as more content and resources get created. For example, Philipp has a [blog post](https://www.philschmid.de/sagemaker-llama2-qlora) about fine-tuning LLaMA 2 on SageMaker here that can be added :)

> Although I think this is already the case, would it be possible for you to sort these incomplete models and provide the top 20 sorted by impact or page views as of recent advances?

By downloads, here are the next top 20 models (its okay to skip some of the models if there aren't any available resources for them):

<details>

BART
CLIPSeg
Marian
MPNet
ELECTRA
ResNet
CamemBERT
HuBERT
LLaMA
Longformer
VisionEncoderDecoder
GPT NeoX
EnCodec
ConvBERT
mBART
GPT Neo
FNet
YOLOS
BLIP
BEiT

</details>

**Comment by ahtashamilyas** [comment]
ok

**Comment by ajaitly11** [comment]
Hello, I would like to put together some resources for Longformer, willing to look into CamemBERT as well if Longformer has already been taken.

**Comment by debrupf2946** [comment]
Hi @stevhliu  can I work on LLaMA, can you please assign me?

**Comment by siddharth1012** [comment]
Hi @stevhliu Can I work on ALBERT, which is the only left model in the list?

**Comment by ENate** [comment]
Hi @siddharth1012 I am almost done with ALBERT and about to open a PR. Issue is due to a doc-builder issue I encountered. Also the problem on Cython version.

**Comment by ENate** [comment]
So I am working on ALBERT thanks

**Comment by stevhliu** [comment]
Hi, thanks for your interest @jaykhatri0875! The open model list is [here](https://github.com/huggingface/transformers/issues/19848) but I think we're close to completing it. If you're interested in making other contributions, feel free to check out the [Good First Issues](https://github.com/huggingface/transformers/issues?q=is:open+is:issue+label:%22Good+First+Issue%22) 🤗

**Comment by stevhliu** [comment]
All finished now! 🥳

**Comment by ENate** [comment]
Hi @stevhliu. Do you think it may be great to modify the docs in such a way that we can have multiple tabs for examples in Jax, Tensorflow or Pytorch or this is not necessary? For example, each tab will consist of examples written in a particular framework?
